### ISSUE TITLE
Container spam/overload

### ISSUE THREAD

**Issue by Jarvvski** [containrrr/watchtower#878]
**Describe the bug**
Many watchtower containers spawning

**To Reproduce**
Unsure, seems random currently. Not sure how it's starting

**Expected behavior**
Not spawn many containers and overwhelm host node

**Screenshots**
![image](https://user-images.githubusercontent.com/6012840/113441688-9cf51280-93e6-11eb-93b5-88991c8a6d13.png)

**Environment**
 - Platform Ubuntu 20.04
 - Architecture x64
 - Docker version 20.10.5

**Logs**

```sh
OaGxILuHEgmnCPAGtYXqyVOIytEpiYOR | standard_init_linux.go:219: exec user process caused: no such file or directory
QGHrodQwaGFwbbEHKCMExwRlBvpDlSXe | standard_init_linux.go:219: exec user process caused: no such file or directory
PgluIhiPwXqwvuxNXGTgJEYtEujXTedm | standard_init_linux.go:219: exec user process caused: no such file or directory
gEMyxvAgDITlmvlUCARkIMclltNeLGpa | time="2021-03-29T05:24:04Z" level=warning msg="Could not do a head request, falling back to regular pull." container=/portainer_agent.ue0iw0j0qwu55cumypljl98ox.r6zsa2crcfpkipuoxsa1l9uey image="portainer/agent:latest@sha256:<snip>"
gEMyxvAgDITlmvlUCARkIMclltNeLGpa | time="2021-03-29T05:27:04Z" level=warning msg="Could not do a head request, falling back to regular pull." container=/portainer_agent.ue0iw0j0qwu55cumypljl98ox.r6zsa2crcfpkipuoxsa1l9uey image="portainer/agent:latest@sha256:<snip>"
KEJAymDpQcmvmyPoVCndtTqUZxIbZKyE | standard_init_linux.go:219: exec user process caused: no such file or directory
xdGjtlPWVjAlczsXcAszBzUbMKXwzfSN | standard_init_linux.go:219: exec user process caused: no such file or directory
MJBvOZmwuxuVhumHertFVIIzMyfhAeco | standard_init_linux.go:219: exec user process caused: no such file or directory
DDOEAObaFFurLdQrwOpIaHCsImyQuGPh | standard_init_linux.go:219: exec user process caused: no such file or directory
XoHoEHVRmzBUjGlQmYURpDQimjpYFNad | standard_init_linux.go:219: exec user process caused: no such file or directory
uWypqYfSRMYkZAAxtTUKYoXyXVDMtmZj | standard_init_linux.go:219: exec user process caused: no such file or directory
zVXlfbGZAaHdMlEOCzekZorzJMZFNAPR | standard_init_linux.go:219: exec user process caused: no such file or directory
zRAsqsCWtBWFnayvTQdQguIVHzxOcmbs | standard_init_linux.go:219: exec user process caused: no such file or directory
watchtower    | standard_init_linux.go:219: exec user process caused: no such file or directory
OhUMzaecfwJYuBocExWdOQvPVpJJPvlL | standard_init_linux.go:219: exec user process caused: no such file or directory
GHgqfrJGYPtvOlpTUXxnUavhSMNZYtvm | standard_init_linux.go:219: exec user process caused: no such file or directory
YleChspSPVwRQHGVBxilQByWZVjrihxs | standard_init_linux.go:219: exec user process caused: no such file or directory
GYuEezBBkDRPAvBUSmllbsCvuuQPzbUh | standard_init_linux.go:219: exec user process caused: no such file or directory
ifvLoKHipkdzrphnVqAkmySAsgqhDKuI | standard_init_linux.go:219: exec user process caused: no such file or directory
UAozLocTySzcWvLNOQJPASaEMNGALenB | standard_init_linux.go:219: exec user process caused: no such file or directory
IZhmfPjeAEsQIqnxHqMaaLaULeOuPlaQ | standard_init_linux.go:219: exec user process caused: no such file or directory
obmjHOYNrUexZBKXKRuffXjdaNOXJJnu | standard_init_linux.go:219: exec user process caused: no such file or directory
mzpGmPFnAtIMdgBFLCduhGoxuVXqHUWt | standard_init_linux.go:219: exec user process caused: no such file or directory
uxGUQKnCiaZKljFjYCGFNFYlWZnYBPyL | standard_init_linux.go:219: exec user process caused: no such file or directory
OaGxILuHEgmnCPAGtYXqyVOIytEpiYOR | standard_init_linux.go:219: exec user process caused: no such file or directory
QGHrodQwaGFwbbEHKCMExwRlBvpDlSXe | standard_init_linux.go:219: exec user process caused: no such file or directory
PgluIhiPwXqwvuxNXGTgJEYtEujXTedm | standard_init_linux.go:219: exec user process caused: no such file or directory
EinNfsaCJTUZtaBObgOffuWuasYbrYQw | standard_init_linux.go:219: exec user process caused: no such file or directory
```

**Additional context**
We run docker swarm, with 3 nodes. Watchtower is on the master node

**Comment by Jarvvski** [comment]
Here's the compose file by the way:

```yaml
version: "3.7"

services:
  watchtower:
    image: containrrr/watchtower
    container_name: watchtower
    restart: unless-stopped
    networks:
      - edge
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    command: --schedule "0 */3 * * *" --cleanup
    logging:
      driver: "json-file"
      options:
        max-size: "5m"
        max-file: "10"

networks:
  edge:
    external: true
    name: edge
```

**Comment by piksel** [comment]
Hm, I don't think it's watchtower that starts itself, but the logs don't tell much. It could be that the broken `v1.2.0` version caused this somehow. If that's the case it won't happen again (unless another broken build is released, of course, but hopefully that won't be the case), but you probably need to stop all running watchtower instances to start it in a "clean" state.
Running `docker pull containrrr/watchtower:latest` might also help.

**Comment by Jarvvski** [comment]
Yeah, I killed all current containers of it and did a docker-compose pull

Hopefully it should be fine now, but I had noticed this happen and was checking I didn't miss any special config. I thought maybe it would auto-scale based on some reasoning, seeing as it had access to the docker socket

Thanks for the quick response!

**Comment by simonaronsson** [comment]
Hey @Jarvvski,

This is indeed an error that was caused by the regression we saw in `v1.2.0`. It's been verified as solved in `v1.2.1`. Sorry for the issues it may have caused you.
